### PR TITLE
Docs/kv state store updates

### PR DESCRIPTION
**Pull request by halsaphi**
## Change Description

Documentation for the state store configuration feature. 

## Change Type
**Select ONE**
- [ x] Documentation

- [ ] Stability bug fix

- [ ] Other

- [ ] Other - special case

## Consensus Changes
- [ ] Consensus Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [ x] Documentation Additions

### CODE REVIEW

File: docs/01_nodeos/02_usage/05_nodeos-implementation.md [hunk 1]
```diff
@@ -10,11 +10,22 @@ Every `nodeos` instance creates some internal files to housekeep the blockchain
...
-* The `chain state` or `chain database` is currently stored and cached in a memory mapped file. It contains the blockchain state associated with each block, including account details, deferred transactions, and data stored using multi index tables in smart contracts. The last 65,536 block IDs are also cached to support Transaction as Proof of Stake (TaPOS). The transaction ID/expiration is also cached until the transaction expires.
+* The `chain state` or `chain database` is stored either in `chainbase` or in `rocksdb`, dependant on the `nodeos` `chain_plugin` configuration option `backing-store`. It contains the blockchain state associated with each block, including account details, deferred transactions, and data stored using multi index tables in smart contracts. The last 65,536 block IDs are also cached to support Transaction as Proof of Stake (TaPOS). The transaction ID/expiration is also cached until the transaction expires.
```
Comment (iamveritas) [inline docs/01_nodeos/02_usage/05_nodeos-implementation.md:13]: `dependant`?... I think `depending`? the first is noun.

File: docs/01_nodeos/02_usage/05_nodeos-implementation.md [hunk 2]
```diff
@@ -10,11 +10,22 @@ Every `nodeos` instance creates some internal files to housekeep the blockchain
...
 
 * The `pending block` is an in memory block containing transactions as they are processed and pushed into the block; this will/may eventually become the head block. If the `nodeos` instance is the producing node, the pending block is distributed to other `nodeos` instances.
 * Outside the chain state, block data is cached in RAM until it becomes final/irreversible; especifically the signed block itself. After the last irreversible block (LIB) catches up to the block, that block is then retrieved from the irreversible blocks log.
 
+### Configurable state storage
+
+Nodeos stores the transaction history and a current state. The transaction history is stored in the blocks.log file on disk. State, which is changed by the execution of transactions, is currently stored using chainbase. EOSIO 2.1 introduces configurable state storage and currently supports these backing stores:
```
Comment (iamveritas) [inline docs/01_nodeos/02_usage/05_nodeos-implementation.md:20]: chainbase -> Chainbase (considering how it is capitalized in the other places where is used)

File: docs/01_nodeos/02_usage/60_how-to-guides/10_how-to-configure-state-storage.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
+# Summary
+This how-to describes configuration of the Nodeos `backing store`. Nodeos can now use chainbase or rocks-db as a backing store for smart contract state.
```
Comment (iamveritas) [inline docs/01_nodeos/02_usage/60_how-to-guides/10_how-to-configure-state-storage.md:2]: alternative: describes the configuration of Nodeos
chainbase, rocks-db -> Chainbase, RocksDB

File: docs/01_nodeos/02_usage/60_how-to-guides/10_how-to-configure-state-storage.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+   
+# Prerequisites
+Version 2.1 or above of the EOSIO development environment. 
+
+# Parameter Definitions 
+Specify which backing store to use with the chain_plugin --backing-store argument. This argument sets state storage to either `chainbase`, the default, or `rocksdb`.
```
Comment (iamveritas) [inline docs/01_nodeos/02_usage/60_how-to-guides/10_how-to-configure-state-storage.md:8]: chainbase, rocksdb - same comment as above we should strive to be consistent how we capitalize it.

**Review by iamveritas — CHANGES_REQUESTED** [review]
a few of minor comments.

File: docs/01_nodeos/02_usage/05_nodeos-implementation.md (hunk 2 above)
Comment (halsaphi) [reply docs/01_nodeos/02_usage/05_nodeos-implementation.md:20]: Case, in this case is based on sentence structure.

File: docs/01_nodeos/02_usage/60_how-to-guides/10_how-to-configure-state-storage.md (hunk 4 above)
Comment (halsaphi) [reply docs/01_nodeos/02_usage/60_how-to-guides/10_how-to-configure-state-storage.md:8]: Case in this case is based on parameter use.

**Comment by halsaphi** [comment]
Changing to branch from develop as per lparisc review.

**Pull request closed without merging**